CreeDictionary/utils/fst_analysis_parser.py

```diff
@@ -8,7 +8,7 @@
 
 
 analysis_pattern = re.compile(
-    r"(?P<category>\+N\+A(\+D)?|\+N\+I(\+D)?|\+V\+AI|\+V\+T[AI]|\+V\+II|(\+Num)?\+Ipc|\+Pron).*?$"
+    r"(?P<category>\+N\+A(\+D(?=\+))?|\+N\+I(\+D(?=\+))?|\+V\+AI|\+V\+T[AI]|\+V\+II|(\+Num)?\+Ipc|\+Pron).*?$"
```

Comment:
UHHHHHHHHH

`fst-lookup` returns the parts for you automatically, without you having to parse the string or need ad hoc, brittle regexps. Could you parsed the analysis _without_ using a regular expression?

Reply:
this is needed because we almost had to use hfstol to import xml files for the efficiency. We can of course always use `fst-lookup` everywhere else for a more robust code.

Reply:
but hey, I have a parametrized unit test for this!

Reply:
I'd like to see this ad hoc regular expression go away in future iterations, because the information that it's trying to extract is already available if we use the FST in a different way. Additionally, the FST can change its labels, and this will cause us nightmares when we have to deal with languages  that are _not_ Cree, e.g., Tsuut'ina, and Haida.

I will not block merge on this, but just FYI.

Reply:
@eddieantonio You need to think of this as an optimization here.

Reply:
Protip: do NOT tell your supervisors what to think.

Regardless if this is an "optimization" or not, it is a liability:

 - the regular expression is coupled with the extrinsic behaviour of the FST analysis string, which has and _can_ **change** as we patch things in the FST
   - this code must be in sync with changes to the FST, which adds a maintainability burden -- we have to remember to change this line if the FST gains new tags or changes its existing tags
 - the regular expression will not support FSTs for other languages, which @aarppe DOES want to see happening in the future
 - the regular expression is difficult to modify, as it is a lot of difficult to read code written all on one string. Have you considered using `re.VERBOSE`?
 - once again, the FST already has this information preparsed! We absolutely do not have to manually maintain a regular expression from scratch when the FST data structure already contains this information.

I give this regex a rating of:
![not stonks](https://user-images.githubusercontent.com/2294397/72646742-17498c00-3934-11ea-8208-52f8379523f0.png)

Reply:
whoa OK. Yes.